enhancements/network/on-prem-service-load-balancers.md

```diff
@@ -229,35 +230,123 @@ doing all of your testing in a virtual network built on top of it.
-additional component to be installed on on-premise clusters.  However, in the
-[ROADMAP.md
+The MetalLB operator will be managed by OLM as an optional additional component
+to be installed on on-premise clusters.  However, in the [ROADMAP.md
 document](https://github.com/openshift/enhancements/blob/master/ROADMAP.md),
 there is an item to "Front the API servers and other master services with
 service load balancers".  If this functionality is required at install time,
-the details on management of this operator may be revisited.
+the details on management of this operator may be revisited, since MetalLB may
+become a hard requirement in some environments.
 
 There is a start of a
 [metallb-operator](https://github.com/cybertron/metallb-operator) available and
 a [video demo](https://www.youtube.com/watch?v=WgOZno0D7nw).
 
-#### Alternative Integration: Cloud Controller Manager
-
-An alternative integration approach would be via a cloud controller manager
-(CCM). An example of this is the [packet.net
-CCM](https://github.com/packethost/packet-ccm), which ensures MetalLB is
-deployed and also configures it properly to work in packet.net’s BGP
-environment.
-
-These integration options must be explored in more detail as part of a more
-detailed integration proposal.
+#### Configuration Interfaces
+
+MetalLB's current configuration interface is a single `ConfigMap`.  We only
+want to support CRD based configuration through OpenShift.  The ultimate goal
+will be to enhance MetalLB with a native CRD based configuration interface.
+The design work has begun here:
+
+https://github.com/metallb/metallb/pull/833
+
+Since this is an upstream effort, the completion of this work may take longer
+than the timeline for when we want to ship the first piece of MetalLB support
+for OpenShift, which will be to support MetalLB in layer2 mode only.
+
+To mitigate this timing risk, we have another option for a way to configure the
+layer2 mode of MetalLB by adding a small new feature to OpenShift's existing
+network config object.  OpenShift has an existing feature to automatically
+assign external IPs to services of `type=LoadBalancer`.  For example, take this
+sample configuration:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  ...
+  externalIP:
+    autoAssignCIDRs:
+    - 10.10.10.0/24
+```
+
+This makes OpenShift automatically set the `externalIP` field, but nothing will
+make sure those addresses are reachable automatically.  What we need is a
+similar option that instead of setting `ExternalIPs`, it should set the
+`LoadBalancerIP` field in the Service spec.  The proposed new field is
+`autoAssignLoadBalancerCIDRs`.  With the new option, the way to configure the
+address pool for MetalLB would be:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  ...
+  externalIP:
+    autoAssignLoadBalancerCIDRs:
```

Comment:
should this field be part of the `ExternalIPConfig`?

Reply:
Thanks for raising this.  It doesn't make sense there.  I moved it to a new section like:

```
    loadBalancer:
        autoAssignCIDRs:
            - ...
```